### PR TITLE
Change references from AddressBook to Database

### LINKED ISSUE #105: Remove more AddressBook references
Some things like `AddressBookParser` doesn't actually got anything to do with the `AddressBook` class, it is used as a product name here, another one is the name of the save file, these are in the same vein as #98 but didn't really want to clutter the already huge PR with more minor details.

### PR DESCRIPTION
**Pull request by ZhangAnli**
Reduce references to "AddressBook" in code and instead replaced with "Database". This is to move away from AB3 implementation and move towards Pawbook implementation in code. 

Closes #105.

### CODE REVIEW

File: src/main/java/dog/pawbook/logic/commands/ExitCommand.java [hunk 1]
```diff
@@ -9,7 +9,7 @@
 
     public static final String COMMAND_WORD = "exit";
 
-    public static final String MESSAGE_EXIT_ACKNOWLEDGEMENT = "Exiting Address Book as requested ...";
+    public static final String MESSAGE_EXIT_ACKNOWLEDGEMENT = "Exiting database as requested ...";
```
Comment (kouyk) [inline src/main/java/dog/pawbook/logic/commands/ExitCommand.java:12]: this should be pawbook?

File: src/main/java/dog/pawbook/storage/AddressBookStorage.java [hunk 2]
```diff
@@ -5,41 +5,42 @@
...
 
     /**
      * Returns the file path of the data file.
      */
-    Path getAddressBookFilePath();
+    Path getDatabaseFilePath();
 
     /**
-     * Returns AddressBook data as a {@link ReadOnlyAddressBook}.
+     * Returns AddressBook data as a {@link ReadOnlyDatabase}.
```
Comment (kouyk) [inline src/main/java/dog/pawbook/storage/AddressBookStorage.java:22]: Still got addressbook here

File: src/test/java/dog/pawbook/storage/JsonAddressBookStorageTest.java [hunk 3]
```diff
@@ -46,34 +46,34 @@ private Path addToTestDataPathIfNotNull(String prefsFileInTestDataFolder) {
...
     @Test
     public void read_missingFile_emptyResult() throws Exception {
-        assertFalse(readAddressBook("NonExistentFile.json").isPresent());
+        assertFalse(readDatabase("NonExistentFile.json").isPresent());
     }
 
     @Test
     public void read_notJsonFormat_exceptionThrown() {
-        assertThrows(DataConversionException.class, () -> readAddressBook("notJsonFormatAddressBook.json"));
+        assertThrows(DataConversionException.class, () -> readDatabase("notJsonFormatAddressBook.json"));
```
Comment (kouyk) [inline src/test/java/dog/pawbook/storage/JsonAddressBookStorageTest.java:54]: do you want to also rename the json files?

File: src/test/java/dog/pawbook/storage/JsonAddressBookStorageTest.java [hunk 4]
```diff
@@ -46,34 +46,34 @@ private Path addToTestDataPathIfNotNull(String prefsFileInTestDataFolder) {
...
 
     @Test
-    public void readAddressBook_invalidAndValidOwnerAddressBook_throwDataConversionException() {
-        assertThrows(DataConversionException.class, () -> readAddressBook("invalidAndValidOwnerAddressBook.json"));
+    public void readDatabase_invalidAndValidOwnerDatabase_throwDataConversionException() {
+        assertThrows(DataConversionException.class, () -> readDatabase("invalidAndValidOwnerAddressBook.json"));
     }
 
     @Test
     public void readAndSaveAddressBook_allInOrder_success() throws Exception {
```
Comment (kouyk) [inline src/test/java/dog/pawbook/storage/JsonAddressBookStorageTest.java:68]: the method names too!

File: src/test/java/dog/pawbook/testutil/DatabaseBuilder.java [hunk 5]
```diff
@@ -0,0 +1,34 @@
...
+
+import dog.pawbook.model.Database;
+import dog.pawbook.model.managedentity.owner.Owner;
+
+/**
+ * A utility class to help with building Addressbook objects.
+ * Example usage: <br>
+ *     {@code AddressBook ab = new AddressBookBuilder().withOwner("John", "Doe").build();}
+ */
+public class DatabaseBuilder {
```
Comment (kouyk) [inline src/test/java/dog/pawbook/testutil/DatabaseBuilder.java:11]: probably need separate this rename as part of another commit so as to track as a rename instead of creation of new file. There's several references that needs to be renamed as well

**Review by kouyk — COMMENTED** [review]
Overall most things are renamed already, just some bits here and there overlooked. But my main worry is that some of the changes on the test side of things might cause issues as charles is currently working on refactoring some of the tests, you might want to check with him to make his life a bit easier.

File: src/main/java/dog/pawbook/logic/commands/ExitCommand.java (hunk 1 above)
Comment (ZhangAnli) [reply src/main/java/dog/pawbook/logic/commands/ExitCommand.java:12]: Edited.

File: src/main/java/dog/pawbook/storage/AddressBookStorage.java (hunk 2 above)
Comment (ZhangAnli) [reply src/main/java/dog/pawbook/storage/AddressBookStorage.java:22]: Edited.

File: src/test/java/dog/pawbook/storage/JsonAddressBookStorageTest.java (hunk 4 above)
Comment (ZhangAnli) [reply src/test/java/dog/pawbook/storage/JsonAddressBookStorageTest.java:68]: Edited.

File: src/test/java/dog/pawbook/storage/JsonAddressBookStorageTest.java (hunk 3 above)
Comment (ZhangAnli) [reply src/test/java/dog/pawbook/storage/JsonAddressBookStorageTest.java]: Edited.

**Review by kouyk — APPROVED** [review]
